### PR TITLE
Checkout save new payment methods

### LINKED ISSUE #629: Checkout: Save payment method
For new payment methods (Card or Bank Account):
- If checkout has a `payment_method_group_id`, show a checkbox labeled "Save payment method"
- If the checkbox is checked, pass the `payment_method_group_id` along with the tokenize `paymentMethodMetadata` (Make sure that PaymentsJS application passes this along appropriately to the backend). See the tokenize call in the `PaymentMethodForm` component.

See [Create Payment Method](https://docs.justifi.tech/api-spec#tag/Payment-Methods) docs and expand `payment_method` section

### PR DESCRIPTION
**Pull request by colganmeanor**
### Add logic to save new payment methods

NOTE: this PR will need to be coordinated with the following PR for payments-js:
[Payments-js 136](https://github.com/justifi-tech/payments-js/pull/136
)
This PR commits the following changes:

- Adds a checkbox below the new payment method billing form that will only render IF the checkout has an associated `payment_method_group_id`
- When checked / true, the checkout will submit the payment_method_group_id along with payment's metadata

Links
-----

Closes #629 

Developer considerations
--------------

- On any task
  - [X] Previous unit and e2e tests are passing
  - [ ] Perform dev QA

Developer QA steps
--------------------
- Make sure that payment_method_group_id is being sent
- Rendering a new checkout shows the saved payment method

### CODE REVIEW

File: packages/webcomponents/src/components/checkout/save-new-payment-method.tsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  handleCheckboxChange(event: Event) {
+    const target = event.target as HTMLInputElement;
+    this.isChecked = target.checked;
+    this.checkboxChanged.emit(this.isChecked);
+  }
+
+  render() {
+    return (
+      <div class="mt-3 d-flex align-items-center">
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/components/checkout/save-new-payment-method.tsx:18]: I tried with `mt-4` and it looked a little bit more centralized. WDYT?

mt-3: 
image

mt-4: 
image

Comment (colganmeanor) [reply packages/webcomponents/src/components/checkout/save-new-payment-method.tsx:18]: Yeah, agreed. It's a little better with `mt-4`. I will update that.

Good catch!

File: packages/webcomponents/src/components/checkout/new-payment-method.tsx [hunk 2]
```diff
@@ -52,7 +62,13 @@ export class NewPaymentMethod {
...
     try {
       const billingFormFieldValues = await this.billingFormRef.getValues();
-      const paymentMethodData = { ...billingFormFieldValues };
+      let paymentMethodData;
+      if (this.saveNewPaymentMethodChecked) {
+        paymentMethodData = { ...billingFormFieldValues, payment_method_group_id: this.paymentMethodGroupId };
+      } else {
+        paymentMethodData = { ...billingFormFieldValues };
+      }
+      console.log('paymentMethodData', paymentMethodData);
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/components/checkout/new-payment-method.tsx:71]: Oops

File: packages/webcomponents/src/components/checkout/new-payment-method.tsx [hunk 3]
```diff
@@ -77,6 +93,7 @@ export class NewPaymentMethod {
         </div>
         <h3 class="fs-6 fw-bold lh-lg mb-4">Billing address</h3>
         <justifi-billing-form ref={(el) => (this.billingFormRef = el)} />
+        <justifi-save-new-payment-method hidden={!this.paymentMethodGroupId} />
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/components/checkout/new-payment-method.tsx:96]: Nice!! I totally forgot about the `hidden` HTML attribute. 👏

File: packages/webcomponents/src/components/checkout/checkout-core.tsx [hunk 4]
```diff
@@ -155,6 +155,7 @@ export class CheckoutCore {
             show-card={!this.disableCreditCard}
             show-ach={!this.disableBankAccount}
             show-bnpl={!this.disableBnpl}
+            paymentMethodGroupId={this.paymentMethodGroupId}
```
Comment (TiagoFuelber) [inline packages/webcomponents/src/components/checkout/checkout-core.tsx:158]: I think we could add the `payment_method_group_id` to the `Checkout` class constructor then we can pass this prop like: 
`paymentMethodGroupId={this.checkout?.payment_method_group_id}`

Reasons:
* Make it consistent with the backend object entity
* Avoid derived state: line 35 and 69

File: packages/webcomponents/src/components/checkout/new-payment-method.tsx (hunk 2 above)
Comment (colganmeanor) [reply packages/webcomponents/src/components/checkout/new-payment-method.tsx:71]: 🙃 🤪

File: packages/webcomponents/src/components/checkout/checkout-core.tsx (hunk 4 above)
Comment (colganmeanor) [reply packages/webcomponents/src/components/checkout/checkout-core.tsx:158]: Ah this is much better. Thankfully it seems we are already passing this in the Checkout constructor - so its simple enough to just remove lines 35 and 69. 😄

**Review by TiagoFuelber — APPROVED** [review]
Nice work man! Approved! 💻

**Review by TiagoFuelber — DISMISSED** [review]
Nice work man! New changes make sense to me. And new payment methods are showing up when reload the checkout after saving it. Approved

**Pull request merged**